vagrant-spk

```diff
@@ -395,6 +395,86 @@ mkdir -p /var/run
 /usr/sbin/nginx -g "daemon off;"
 """
 
+DIY_SETUP_SCRIPT = r"""#!/bin/bash
+# This script is run in the VM once when you first run `vagrant-spk up`.  It is
```

Comment:
You make an interesting point about the similarity between `setup.sh` and `build.sh` -- I suspect the separation was a result of the way that I think about the separation between platform and app, but in this case the OS dependencies can be thought of as just part of the app as well.  Perhaps they should be merged into a single "prepare the VM to run your code" build step.

One counterpoint here is that scripts are probably bad at avoiding redoing work -- I don't really want to force people to wait for an `apt-get update` to complete every time they run `vagrant-spk dev`.  On the other hand, I don't want someone trying to get a clean build to wind up with old, cached artifacts.  Perhaps this is why Docker went with the "cache the FS after each command" approach, but I dislike imperfect caching that users aren't explicitly aware of, i.e. old version of libssl package was cached and subsequent build will use the cache and still be vulnerable.